omnigibson/robots/manipulation_robot.py

```diff
@@ -986,7 +1009,7 @@ def _establish_grasp_rigid(self, arm="default", ag_data=None):
 
         # Create a p2p joint if it's a child link of a fixed URDF that is connected by a revolute or prismatic joint
         joint_type = "FixedJoint"
-        if ag_obj.fixed_base:
+        if ag_obj.fixed_base or ag_obj.root_link != ag_link:
```

Comment:
I think the logic here is a bit confusing.

Can we go through the following three scenarios and explicitly write comments for each of them?
1. when should we create a fixed joint?
2. when should we create a spherical joint?
3. when should we NOT create any joint? (this is related to `can_assisted_grasp`) 

Objects to consider:
1. fixed/not fixed
2. children links / root links
3. heavy / light

Maybe we should put all of these logics in one place (instead of in two different places currently) and add detailed comments. Otherwise, it's very easy to forget.

Reply:
I have combined this logic. Might be best to review the commit. https://github.com/StanfordVL/OmniGibson/commit/df170c2e2d019c6b9905790e37273266004e065d